cerbero/tools/strip.py

```diff
@@ -18,33 +18,53 @@
 # Boston, MA 02111-1307, USA.
 
 import os
+import asyncio
 from cerbero.config import Platform
-from cerbero.utils import shell
+from cerbero.utils import shell, messages as m
+
 
 class Strip(object):
     '''Wrapper for the strip tool'''
 
-    STRIP_CMD = '$STRIP'
-
     def __init__(self, config, excludes=None, keep_symbols=None):
         self.config = config
         self.excludes = excludes or []
         self.keep_symbols = keep_symbols or []
+        self.strip_cmd = os.environ['STRIP']
+
+    async def async_strip_file(self, path):
+        if not self.strip_cmd:
+            m.warning('Strip command is not defined')
+            return
 
-    def strip_file(self, path):
         for f in self.excludes:
             if f in path:
                 return
+
+        if self.config.target_platform == Platform.DARWIN:
+            cmd = [self.strip_cmd, '-x', path]
+        else:
+            cmd = [self.strip_cmd]
+            for symbol in self.keep_symbols:
+                cmd += ['-K', symbol]
+            cmd += ['--strip-unneeded', path]
+
         try:
-            if self.config.target_platform == Platform.DARWIN:
-                shell.call("%s -x %s" % (self.STRIP_CMD, path))
-            else:
-                shell.call("%s %s --strip-unneeded %s" % (self.STRIP_CMD,
-                    ' '.join(['-K %s' % x for x in self.keep_symbols]), path))
-        except:
-            pass
+            await shell.async_call(cmd)
+        except Exception as e:
+            m.warning(e)
+
+    def strip_file(self, path):
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(self.async_strip_file(path))
 
     def strip_dir(self, dir_path):
+        if not self.strip_cmd:
+            m.warning('Strip command is not defined')
+            return
+
+        tasks = []
         for dirpath, dirnames, filenames in os.walk(dir_path):
             for f in filenames:
-                self.strip_file(os.path.join(dirpath, f))
+                tasks.append(self.async_strip_file(os.path.join(dirpath, f)))
```

Comment:
This is a bit dangerous, you are parallelizing without limiting the numbers of parallel tasks. If you have 4000 files you will run 4000 process all together.

Reply:
This is exactly what shell.run_until_complete(tasks) does. It limits by default to 16 `max_concurrent=16` concurrent tasks.